source/common/network/listener_impl.h

```diff
@@ -8,6 +8,10 @@
 #include "common/event/dispatcher_impl.h"
 #include "common/event/libevent.h"
 
+#include "server/connection_handler.h"
```

Comment:
In this case I can actually get away with using a forward reference to avoid the dependency.
Otherwise, if ConnectionHandler abstract class becomes part of common, the issue goes away.